packages/snaps-controllers/src/snaps/SnapController.ts

```diff
@@ -1466,17 +1466,20 @@ export class SnapController extends BaseController<
       'PermissionController:getPermissions',
       origin,
     ) as SubjectPermissions<PermissionConstraint>;
-    const snapIdsCaveat = (subjectPermissions?.[
+    const snapIdsCaveat = subjectPermissions?.[
       WALLET_SNAP_PERMISSION_KEY
-    ]?.caveats?.find((caveat) => caveat.type === SnapCaveatType.SnapIds) ??
-      {}) as Caveat<string, Json>;
+    ]?.caveats?.find((caveat) => caveat.type === SnapCaveatType.SnapIds) as
+      | Caveat<string, Json>
+      | undefined;
+
+    assert(snapIdsCaveat !== undefined);
 
     const caveatHasSnap = Boolean(
-      (snapIdsCaveat.value as Record<string, unknown>)?.[snapId],
+      (snapIdsCaveat.value as Record<string, Json>)?.[snapId],
```

Comment:
```suggestion
      snapId in snapIdsCaveat.value,
```

Reply:
That doesn't work because value is of type `Json`, can't index it with type `string` so you have to cast it.

Reply:
```
> 42?.foo
undefined
> 42.foo
42.foo
^^^

Uncaught SyntaxError: Invalid or unexpected token
```
🤔 wat

I'd prefer not force casting `Json` into `Record<>` rather than `snapIdsCaveat.value?.[snapId] as Json | undefined`, but it's good enough since it'll work in runtime.

Reply:
I know it might not be the cleanest, but I think it's a fair assumption that the caveat value is the shape it is once we're at the point of disconnecting a snap. At least until we improve typing for snap caveats in general, I think.

TS still has the same complaint with your suggestion, whether or not value is defined doesn't matter -- if it is defined than TS complains that we're assuming the shape of the `Json` object by trying to index with a `string`, which it is correct to do so.